### ISSUE TITLE
Convenient way to obtain the plotted elements

### ISSUE THREAD

**Issue by Datseris** [bifurcationkit/BifurcationKit.jl#84]
I prefer to plot with Makie, which in my humble opinion is a plotting framework vastly superior to Plots.jl. I wish in the future BifurcationKit.jl also moves to Makie. But that's just a wish for now. 

The difficulty I have right now, at least according with the online tutorial, is that plotting is fully hard-baked into the system. Would it be possible to expose, in a simple way, how to get the actual plotted data, i.e., the stable and unstable curve segments, and the scatter-points used to denote the bifurcations? The `br` object contains this information, but at least the introductory tutorial doesn't show how to obtain it (https://bifurcationkit.github.io/BifurcationKitDocs.jl/stable/tutorials/ode/tutorialsODE/#Neural-mass-equation-(Hopf-aBS))

**Comment by Datseris** [comment]
Alright, I have found out that this information exists here: https://bifurcationkit.github.io/BifurcationKitDocs.jl/stable/library/#BifurcationKit.ContResult  but I would be very glad to see a tutorial of how to use this information, as the docstring is dense with information but without any examples.

**Comment by rveltz** [comment]
> I prefer to plot with Makie, which in my humble opinion is a plotting framework vastly superior to Plots.jl

I agree with you. I have some functions to plot in Makie, but this is less developed than my Plots recipes. 

> is that plotting is fully hard-baked into the system.

I would like to remove it from BifurcationKit and create its own package like `BifurcationKitPlots`. This would remove the `Requires` in the loading and make the switch to Makie easier. The only think I am missing is an interface, or maybe a backend. I could have the backend `plots(), makie(), none()`.

**Comment by Datseris** [comment]
In Julia 1.9 there is no reason for requires nor a reason for independent plotting packages due to weak dependencies ;)

p.s.: I am playing around with the fields of the results struct. I think a convenience function that just returns the curve segments followed by true/false (for stable/unstable) would be much appreciated!

**Comment by rveltz** [comment]
> I would be very glad to see a tutorial of how to use this information, as the docstring is dense with information but without any examples.

I could add this in https://bifurcationkit.github.io/BifurcationKitDocs.jl/stable/plotting/

What you require is the codim1 plot. The recipe is complicated because, I want to plot many curves at ones, filter the bifurcation points, etc. There is also the codim2, bifurcation diagrams, etc. 

In any case, this is https://github.com/bifurcationkit/BifurcationKit.jl/blob/master/src/plotting/Recipes.jl#L74. 

Basically:

`plot(br.param, br.x)`

stability is `br.stable`

**Comment by rveltz** [comment]
> returns the curve segments followed by true/false (for stable/unstable) would be much appreciated!

Hum, what is better than `br.stable`? 

Please note that `br.branch` is a StructArray which makes it easy to access the fields.
You can have an idea of the propertynames by looking at the first point on the branch `br[1]`. Lastly, these propertynames are not fixed: they are changed by the user (for example the parameter axis, or the tuple from `recordFromSolution`). Further, when you branch to codim2 or to periodicorbits, other fields are added like the period of the cycle, etc.

**Comment by Datseris** [comment]
great, thanks, i was reinventing the wheel by doing 
```
n = length(br)
params = [getSolp(br, k) for k in 1:n]
Es = [getSolx(br, k)[1] for k in 1:n]
isstable = [real(eigenvals(br, k)[1]) < 0 for k in 1:n-1]

```

but `br.param, br.E` was what I needed!!!

**Comment by rveltz** [comment]
> isstable = [real(eigenvals(br, k)[1]) < 0 for k in 1:n-1]

Note that this is "wrong". There is a tolerance on the stability (because the eigenvalues may be computed up to a given precision by iterative methods), you should really use `br.stable`.

**Comment by Datseris** [comment]
I re-opened because I couldn't get the scatter points of the bifurcaiton points, but they are actually `br.specialpoint`.

**Comment by Datseris** [comment]
I'll keep this open. It appears that a convenience function that returns the actually plotted information is need. For example, I am continuing with the basic neural mass documentation example. Once I do:
```julia
br_potrap = continuation(
	br, 4, opts_po_cont,
	PeriodicOrbitTrapProblem(M = Mt, jacobian = :Dense, updateSectionEveryStep = 0);
	verbosity = 0, plot = false,
	args_po...,
	callbackN = BifurcationKit.cbMaxNorm(1000.)
)
```

this `br_potrap` is a different object and I can't get the continuation curves from it the same way I did for the `br` object before, with a simple `br.param` for example.

In general, by using BifurcationKit.jl for a little, it appears to me that the current design is that the "holy grail" is the plot. I would argue instead that a design where the "holy grail" is the plotted _information_, instead of the plot itself, is much more user friendly. It takes me 2 lines of code to go from the plotted information to the plot, and the plotted information is what I would actually use if I wanted to do any kind of further analysis.

**Comment by Datseris** [comment]
When I plot `br_potrap.param, br_potrap.max` as you suggest I get:

![image](https://user-images.githubusercontent.com/19669089/230073345-2a929201-8028-40fa-95e1-59b90715de5d.png)

(i've made dashed the parts wjere `br_potrap.stable` is false)

which is different from the online automated call to `plot`

![image](https://user-images.githubusercontent.com/19669089/230073411-094f2722-6462-4c19-a721-e968cae7c845.png)

as you can see, several pieces of information are missing. I want to obtain these pieces of information as actual information, not as a plot.

**Comment by rveltz** [comment]
The tutorial plots more than the default:

```
scene = plot(br, br_potrap, markersize = 3)
plot!(scene, br_potrap.param, br_potrap.min, label = "")
```
it adds the minimum over one period

**Comment by Datseris** [comment]
Notice that even though I compute exactly the same thing as the online tutorial, when I do the plot you recommend I don't get this spiral:

![image](https://user-images.githubusercontent.com/19669089/230078941-a937e204-f943-47de-9e02-f7f8ba334ab9.png)


![image](https://user-images.githubusercontent.com/19669089/230078958-86b24390-a20b-447b-95b5-69275503cd66.png)

---

I am also missing all the scattered individual points. They are not the points in `br_potrap.γ.specialpoint`, or at least the point `nd` that is plotted is not existing there.

**Comment by Datseris** [comment]
To be precise, I don't compute **exactly the same** as online. I've removed aall code related to plotting. My full code is:
```julia
optn_po = NewtonPar(verbose = false, tol = 1e-8,  maxIter = 8)

# continuation parameters
opts_po_cont = ContinuationPar(
	dsmax = 0.1, ds= -0.0001, dsmin = 1e-4, pMax = 0., pMin=-5.,
	maxSteps = 110, newtonOptions = (@set optn_po.tol = 1e-7),
	nev = 3, tolStability = 1e-8, detectBifurcation = 3, saveSolEveryStep=1
)

# arguments for periodic orbits
args_po = (
	recordFromSolution = (x, p) -> begin
	xtt = BifurcationKit.getPeriodicOrbit(p.prob, x, @set par_tm.E0 = p.p)
	return (max = maximum(xtt[1,:]),
			min = minimum(xtt[1,:]),
			period = getPeriod(p.prob, x, @set par_tm.E0 = p.p))
	end,
	normC = norminf
)

Mt = 200 # number of time sections
@time br_potrap = continuation(
	# we want to branch form the 4th bif. point
	br, 4, opts_po_cont,
	# we want to use the Trapeze method to locate PO
	# this jacobian is specific to ODEs
	# it is computed using AD and
	# updated inplace
	PeriodicOrbitTrapProblem(M = Mt, jacobian = :Dense, updateSectionEveryStep = 0);
	# regular continuation options
	verbosity = 0, plot = false,
	args_po...,
	callbackN = BifurcationKit.cbMaxNorm(1000.)
)
```

**Comment by rveltz** [comment]
You might not compute the spiral for some reason

**Comment by rveltz** [comment]
Thanks for your input

**Comment by rveltz** [comment]
Can we close this? The docs have been updated and the Makie recipes as well. See https://github.com/bifurcationkit/BifurcationKit.jl/issues/110
